File: 7-8 kyu/8-Square(n)-Sum.py
# Complete the square sum function so that 
# it squares each number passed into it and then sums the results together.


# ===================================
# 1) One solution
# ===================================


def square_sum(numbers):
    if len(numbers) == 0:
    	return 0
    return sum([pow(c, 2) for c in numbers])



# ===================================
# 2) With a new variable
# ===================================


def square_sum(numbers):
	res = 0
	for num in numbers:
   		res += num*num
	return res

File: 7-8 kyu/8-Square(n)-Sum.js
/*
Complete the square sum function so that 
it squares each number passed into it and then sums the results together.
*/


/*
===================================
1) One solution
===================================
*/

function squareSum(numbers) {
  let sum = 0;
  for( let i = 0; i < numbers.length; i++) {
      sum += Math.pow(numbers[i], 2);
  }
  return sum;
}


/*
===================================
2) With map/reduce
===================================
*/

function squareSum(numbers) {
  return numbers.map(n => n ** 2).reduce((a, b) => a + b, 0);
}


